### ISSUE TITLE
Unreachable reached errror

### ISSUE THREAD

**Issue by bkamins** [JuliaLang/julia#23675]
The following line
https://github.com/bkamins/EventSimulation.jl/blob/master/test/core.jl#L46
produces *Unreachable reached* error on Julia latest (on CI builds and on my local machine), e.g.:
https://ci.appveyor.com/project/bkamins/eventsimulation-jl/build/job/nysdx1xa5xl9q61t
on Julia latest.
The code works fine under Julia 0.6.
The whole repository allowing to reproduce the problem is https://github.com/bkamins/EventSimulation.jl. I am using pure Julia so even if the bug is caused by my programming mistake I would expect it not to thow low-level error.

**Comment by bkamins** [comment]
It seems that the problem is caused by using a global variable as wrapping variable `s` along with the loop in a function like this:
```
who = [1:5;]
v2 = []
function f()
    s = Scheduler()
    for x in randperm(10)
        bulk_register!(s, who, (a,b) -> push!(v2, (s.now, b)), Float64(x), false)
    end
end
f()
```
makes things work again.

The relevant original code starts in the line
https://github.com/bkamins/EventSimulation.jl/blob/master/test/core.jl#L42

**Comment by bkamins** [comment]
Seems to have been fixed by solving #23685
